### PR TITLE
Account app

### PR DESCRIPTION
**Pull request by kuchichan**
- [x] user endpoint & model 
- [x] password retype validation
- [x] Accurate permissions
- [x] Tests
- [x] email activation

This is just for learning pursposes, normally we will use djoser.

### CODE REVIEW

File: coffee_roaster_market/apps/account/serializers.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        model = UserModel
+        fields = ["username", "email", "password", "password_retype"]
+
+    def validate(self, attrs):
+        re_password = attrs.pop("password_retype")
+        if attrs["password"] != re_password:
+            raise serializers.ValidationError("password mismatch!")
+        return super().validate(attrs)
+
+    def create(self, validated_data):
```
Comment (AleksanderPawlak) [inline coffee_roaster_market/apps/account/serializers.py]: Please add type hints.

File: coffee_roaster_market/apps/account/permissions.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from rest_framework import permissions
+
+
+class CurrentUserOrAdmin(permissions.IsAuthenticated):
+    def has_object_permission(self, request, _, obj):
```
Comment (AleksanderPawlak) [inline coffee_roaster_market/apps/account/permissions.py]: Same as: https://github.com/AleksanderPawlak/coffee-roaster-market/pull/12#discussion_r654963786

File: coffee_roaster_market/apps/account/views.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+from .serializers import UserActivationSerializer, UserSerializer, UserCreateSerializer
+from .permissions import CurrentUserOrAdmin
+
+user = get_user_model()
+
+
+class ApiUsersView(viewsets.ModelViewSet):
+    queryset = user.objects.all()
+
+    def get_serializer_class(self):
```
Comment (AleksanderPawlak) [inline coffee_roaster_market/apps/account/views.py]: Same as https://github.com/AleksanderPawlak/coffee-roaster-market/pull/12#discussion_r654963786

File: coffee_roaster_market/apps/account/serializers.py (hunk 1 above)
Comment (kuchichan) [reply coffee_roaster_market/apps/account/serializers.py]: Done.

File: coffee_roaster_market/apps/account/permissions.py (hunk 2 above)
Comment (kuchichan) [reply coffee_roaster_market/apps/account/permissions.py]: Add in entire file.

File: coffee_roaster_market/apps/account/views.py (hunk 3 above)
Comment (kuchichan) [reply coffee_roaster_market/apps/account/views.py]: Add in entire file.

**Review by AleksanderPawlak — APPROVED** [review]

**Pull request merged**